src/log.jl

```diff
@@ -49,26 +51,53 @@ end
 
 let
 global xlog
-const c8 = 0.148197055177935105296783
-const c7 = 0.153108178020442575739679
-const c6 = 0.181837339521549679055568
-const c5 = 0.22222194152736701733275
-const c4 = 0.285714288030134544449368
-const c3 = 0.399999999989941956712869
-const c2 = 0.666666666666685503450651
-const c1 = 2
 
+"""
+First we split the argument to its mantissa `m` and integer exponent `e` so that
+`d = m \times 2^e`,  where `m \in [0.5, 1)` then we apply the polynomial approximant on this 
+reduced argument `m` before putting back the exponent in. 
+This first part is done with the help of the private function `ilogbp1(x)` and we put the exponent back using
+    `\log(m \times 2^e) = \log(m) + \log 2^e = \log(m) + e\times LN2
+
+The polynomial we evaluate is based on coefficients from 
+    `log_2(x) = 2\sum_{n=0}^\infty \frac{1}{2n+1} \bigl(\frac{x-1}{x+1}^{2n+1}\bigr)`
+That being said, since this converges faster when the argument is close to 1, we
+multiply  `m` by `\sqrt(2)/2` and use this in the polynomial approximant instead.  This is done by first 
+first multiplying the argument `d` by `SQRT22`  
+"""
+
+const c8d = 0.148197055177935105296783
+const c7d = 0.153108178020442575739679
+const c6d = 0.181837339521549679055568
+const c5d = 0.22222194152736701733275
+const c4d = 0.285714288030134544449368
+const c3d = 0.399999999989941956712869
+const c2d = 0.666666666666685503450651
+const c1d = 2
+
+const c5f = 0.2371599674224853515625f0
+const c4f = 0.285279005765914916992188f0
+const c3f = 0.400005519390106201171875f0
+const c2f = 0.666666567325592041015625f0
+const c1f = 2.0f0
+
+# sqrt(2)/2
+const SQRT22 = 7.07106781186547524400844362104849039284835937688474036588339868995366239231051e-01
+
+@inline _xlog(x::Float64) = @horner(x, c1d, c2d, c3d, c4d, c5d, c6d, c7d, c8d)
+@inline _xlog(x::Float32) = @horner(x, c1f, c2f, c3f, c4f, c5f)
```

Comment:
This is really throwing off code_native not sure why and generating very bad native code.... any ideas?